src/Common/Crypto/OpenSSLInitializer.cpp

```diff
@@ -63,10 +64,10 @@ void OpenSSLInitializer::initialize()
 #endif
 }
 
-OpenSSLInitializer::~OpenSSLInitializer()
+void OpenSSLInitializer::cleanup()
 {
 #if USE_SSL
-    if (--ref_count == 0)
+    if (cleanup_ref_count++ == 0)
```

Comment:
We want the code in `cleanup` to run only once, right? I think we could use a simple atomic bool `cleanup_done` etc. instead of a full-blown refcount. I guess the same can be said for initialization (`initialization_done`). Whenever cleanup or initialization is called and `*_done` is unexpectedly true, we could throw.